### DISCUSSION TITLE
Dolby Vision RPU not found for POC

### DISCUSSION THREAD

**Discussion by jriker1** [quietvoid/dovi_tool#279] [Q&A]
Use dovi_tool for a lot of dolby vision content.  I basically export the HEVC from the BlueRay, do my video cleanup on the HEVC file, then re-encode with ffmpeg injecting the dolbyvision content again.  This time I received the message in the subject over and over during encoding.

```
frame=208959 fps= 11 q=-0.0 size=5077680624kB time=02:25:15.62 bitrate=4772620.0kbits/s speed=0.445x x265 [warning]: Dolby Vision RPU not found for POC 208955
x265 [warning]: Dolby Vision RPU not found for POC 208956
x265 [warning]: Dolby Vision RPU not found for POC 208957
x265 [warning]: Dolby Vision RPU not found for POC 208958
x265 [warning]: Dolby Vision RPU not found for POC 208959
x265 [warning]: Dolby Vision RPU not found for POC 208960
x265 [warning]: Dolby Vision RPU not found for POC 208961
x265 [warning]: Dolby Vision RPU not found for POC 208962
x265 [warning]: Dolby Vision RPU not found for POC 208963
x265 [warning]: Dolby Vision RPU not found for POC 208964
x265 [warning]: Dolby Vision RPU not found for POC 208965
x265 [warning]: Dolby Vision RPU not found for POC 208966
x265 [warning]: Dolby Vision RPU not found for POC 208967
x265 [warning]: Dolby Vision RPU not found for POC 208968
frame=208973 fps= 11 q=-0.0 size=5078020825kB time=02:25:16.20 bitrate=4772620.0kbits/s speed=0.445x x265 [warning]: Dolby Vision RPU not found for POC 208969
x265 [warning]: Dolby Vision RPU not found for POC 208970
x265 [warning]: Dolby Vision RPU not found for POC 208971
x265 [warning]: Dolby Vision RPU not found for POC 208972
x265 [warning]: Dolby Vision RPU not found for POC 208973
x265 [warning]: Dolby Vision RPU not found for POC 208974
x265 [warning]: Dolby Vision RPU not found for POC 208975
```

Basic commands:

`dovi_tool -m 2 extract-rpu "The Matrix- Resurrections_t00_track1_[eng].hevc"`

then do some neatvideo cleanup in Premiere.  export as MOV.  Then run the below:

`ffmpeg -i "The Matrix- Resurrections_t00.mkv" -sn -an -f yuv4mpegpipe -strict -1 -pix_fmt yuv420p10le - | x265-10b - --input-depth 10 --output-depth 10 --y4m --preset slow --hdr10 --hdr10-opt --high-tier --repeat-headers --crf 17 --master-display "G(13250,34500)B(7500,3000)R(34000,16000)WP(15635,16450)L(40000000,50)" --max-cll "1044,648" --chromaloc 2 --no-sao --range limited --keyint 24 --colormatrix bt2020nc --colorprim bt2020 --transfer smpte2084 --dolby-vision-rpu rpu.bin --dolby-vision-profile 8.1 --vbv-bufsize 160000 --vbv-maxrate 160000 Matrix2021-converted.hevc`

Again do this same process over and over for content first time I think I've seen this.

**Comment by quietvoid** [comment]
Your RPU is clearly incomplete if it's missing metadata for frames, not a `dovi_tool` issue.
The source file is probably invalid.

**Comment by jriker1** [comment]
Looks like it created an 13k file but there is no metadata for any of the frames or the frames a CMD prompt will buffer.  Not sure if there is a way to interrogate the file and see whats in it.

**Comment by quietvoid** [comment]
`dovi_tool info -s RPU.bin`

**Comment by jriker1** [comment]
Thanks yes something must be wrong.  When I extract the MKV from the BD content with MakeMKV dovi_tool extracts like 71k of frames which is maybe a 1/3 of the video?  If I use tsmuxer it returns nothing.

**Comment by quietvoid** [comment]
Is your BD created by tsMuxer? There's a known issue with MakeMKV causing this.

**Comment by jriker1** [comment]
It is basically the original m2ts off the Blue Ray that I exported with MakeMKV.
